k2/csrc/properties.cc

```diff
@@ -33,18 +33,16 @@ bool IsValid(const Fsa &fsa) {
     } else {
       // every state contains at least one arc.
```

Comment:
This is an issue in previous code, but RE "only epsilon arcs enter the final state", this is not right, it should be kFinalSymbol (-1).  You may have to change other code.  This change is necessary, things will break later if you leave it as epsilon.

Reply:
RE "every state contains at least one arc": firstly, you should word it as "every state has at least one arc leaving it".  But actually this condition is not required for an FSA to be valid (it would just not be connected).

Reply:
.. also you should probably make it so that this code can't segfault even if the input is invalid.